lib/workers/repository/finalise/index.ts

```diff
@@ -13,4 +14,7 @@ export async function finaliseRepo(
   await platform.ensureIssueClosing(
     `Action Required: Fix Renovate Configuration`
   );
+  if (config.platformCommit) {
+    await purgeFunnyRenovateRefs();
```

Comment:
We can aggregate ref names and remove them in bulk at the repo finalizing.

However, if finalization wasn't done due to external condition, those refs are lost and won't be removed next run. Since we still want to avoid cluttering, there should be some "purge all" procedure being run periodically. Not sure which schedule is best, but I think run 1 time out of 10–30 would be okay.

Reply:
But can't it be every time we commit?

Reply:
Can I use reverse counter stored in repo cache for this?

Reply:
So, here is optimization I suggest:
- Use private `remoteRenovateRefs` array
- If run counter is zero, populate `remoteRenovateRefs` with remote `refs/renovate/*` refs via `ls-remote`
- In the normal run, decrement counter and initialize `remoteRenovateRefs` with `[]`
- At the finalizing phase, if `remoteRenovateRefs` isn't empty, purge these refs

Reply:
I'm happy as long as it's reasonably performant and doesn't cause unnecessary clones